packages/didcomm/src/modules/connections/DidExchangeProtocol.ts

```diff
@@ -284,32 +286,33 @@ export class DidExchangeProtocol {
     const didDocument = await createPeerDidFromServices(agentContext, services, numAlgo)
     const message = new DidExchangeResponseMessage({ did: didDocument.id, threadId })
 
+    // DID Rotate attachment should be signed with invitation keys
+    const invitationRecipientKeys = outOfBandRecord.outOfBandInvitation
+      .getInlineServices()
+      .map((s) => s.recipientKeys)
+      .reduce((acc, curr) => acc.concat(curr), [])
+
+    // In case the DID Exchange started with an implicit invitation, take all recipient keys from the corresponding DID
+    for (const did of outOfBandRecord.outOfBandInvitation.getDidServices()) {
+      invitationRecipientKeys.push(
+        ...(await getDidDocumentForCreatedDid(agentContext, parseDid(did).did)).recipientKeys.map(
```

Comment:
Do we always need to do this, or only for implicit? Because now we run this always?

Reply:
I think the comment is misleading. It is useful for resolvable DIDs in general.